R/shortTermChange.R

```diff
@@ -0,0 +1,189 @@
+#' (Default: \code{FALSE})
+#' 
+#' @param compositional \code{Logical scalar}. Whether to transform counts.
+#' (Default: \code{FALSE})
+#' 
+#' @param depth \code{Integer scalar}. Specifies the depth used in rarefying. 
+#' (Default: \code{min(assay(x, assay.type)}))
+#' 
+#' @param plot \code{Logical scalar}. Whether to plot short term change.
+#' (Default: \code{FALSE})
+#' 
+#' @param ... additional arguments.
+#' 
+#' 
+#' @return \code{dataframe} or \code{plot} with \code{short term change} 
+#' calculations.
+#' 
+#' @details This approach is used by Wisnoski NI and colleagues
+#'          \url{https://github.com/nwisnoski/ul-seedbank}. Their approach is based on
+#'          the following calculation log(present abundance/past abundance).
+#'          Also a compositional version using relative abundance similar to
+#'          Brian WJi, Sheth R et al
+#'          \url{https://www.nature.com/articles/s41564-020-0685-1} can be used.
+#'          This approach is useful for identifying short term growth behaviors of taxa.
+#'          
+#' @name shortTermChange
+#' 
+#' 
+#' @examples
+#' 
+#' # Load time series data
+#' data(minimalgut)
+#' tse <- minimalgut
+#' 
+#' short_time_labels <- c("74.5h", "173h", "438h", "434h", "390h")
+#' 
+#' # Subset samples by Time_lable and StudyIdentifier
+#' tse <- tse[, !(colData(tse)$Time_label %in% short_time_labels)]
+#' tse <- tse[, (colData(tse)$StudyIdentifier == "Bioreactor A")]
+#' 
+#' # Plot short term change in abundance
+#' shortTermChange(tse, rarefy = TRUE, plot = TRUE) + ggtitle("Bioreactor A")
+
+
+#' @rdname shortTermChange
+#' @export
+setGeneric("shortTermChange", signature = c("x"),
+    function(x,assay.type = "counts", rarefy = FALSE, compositional = FALSE, 
+        depth = min(assay(x, assay.type)), ...)
+        standardGeneric("shortTermChange"))
+
+#' @rdname shortTermChange
+#' @export
+#' @importFrom dplyr arrange as_tibble summarize
+#' @importFrom ggplot2 ggplot aes
+#' @importFrom ggrepel geom_text_repel
+#' @importFrom mia rarefyAssay transformAssay
+#' @importFrom SummarizedExperiment colData
+setMethod("shortTermChange", signature = c(x = "SummarizedExperiment"),
+    function(x, assay.type = "counts", rarefy = FALSE, compositional = FALSE, 
+        depth = min(assay(x, assay.type)), plot = FALSE, ...){
+        ############################## Input check #############################
+        # Check validity of object
+        if(nrow(x) == 0L){
+            stop("No data available in `x` ('x' has nrow(x) == 0L.)",
+                 call. = FALSE)
+        }
+        # Check assay.type
+        mia:::.check_assay_present(assay.type, x)
+        if(!mia:::.is_a_bool(rarefy)){
+            stop("'rarefy' must be TRUE or FALSE.", call. = FALSE)
+        }
+        if(!mia:::.is_a_bool(compositional)){
+            stop("'compositional' must be TRUE or FALSE.", call. = FALSE)
+        }
+        # Ensure that the provided depth is valid
```

Comment:
Instead, these input check functions should be imported in utils.R so that we can use them without mia:::

Reply:
done.